src/main/java/controller/TaskToEventController.java

```diff
@@ -0,0 +1,57 @@
+package main.java.controller;
+
+import main.java.entity.Task;
+import main.java.use_case.TaskToEvent;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class TaskToEventController implements TaskToEventAutoController, TaskToEventManualController {
+    private final TaskToEvent taskToEvent = new TaskToEvent();
+
+    protected final EventController eventController;
+
+    public TaskToEventController(EventController eventController) {
+        this.eventController = eventController;
+    }
+
+    /**
+     * Suggest a time to the user until the user is agrees with the time
+     * @param task the task to be scheduled to event
+     * @return whether the task is successfully scheduled to event
+     */
+    @Override
+    public boolean suggestTimeToUser(Task task) {
+        List<LocalDateTime> unwantedTimes = new ArrayList<>();
+
+        String response;
+        LocalDateTime suggestedTime;
+        Scanner scanner = new Scanner(System.in);
+
+        do {
+            suggestedTime = taskToEvent.getAvailableTime(
+                    task, eventController.calendarData.getCalendar(), eventController.eventScheduler, unwantedTimes
+            );
+            System.out.println("Suggested time: " + suggestedTime);
+            System.out.print("Type 'y' for yes, anything else for no: ");
+            response = scanner.nextLine();
+            // will be effective the next time the loop iterates
+            unwantedTimes.add(suggestedTime);
+        } while (!"y".equals(response));
+
+        return eventController.createEvent(task.getTaskName(), suggestedTime, task.getTimeNeeded());
```

Comment:
Isn't this mutation? We want this method to just be a query. The controller should use the eventScheduler to create an event dependent upon this returned value (this is a concept called command/query separation)